### PR TITLE
Preload settings and templates permissions for post editor

### PR DESCRIPTION
**Pull request by Mamaduka**
PR is a path for following changes in the Gutenberg plugin:

* https://github.com/WordPress/gutenberg/pull/42209
* https://github.com/WordPress/gutenberg/pull/42413

Gutenberg tracking issue: https://github.com/WordPress/gutenberg/issues/43440
Trac ticket: https://core.trac.wordpress.org/ticket/56467

---
**This Pull Request is for code review only. Please keep all other discussion in the Trac ticket. Do not merge this Pull Request. See [GitHub Pull Requests for Code Review](https://make.wordpress.org/core/handbook/contribute/git/github-pull-requests-for-code-review/) in the Core Handbook for more details.**

### CODE REVIEW

**Comment by Mamaduka** [comment]
Sidenote: I really like the suggestion @TimothyBJacobs made recently in `canUser` optimization PR - https://github.com/WordPress/gutenberg/pull/43480#issuecomment-1231013981. Once those suggestions are implemented, I think we can trim down the preloaded paths array.

**Review by ockham — APPROVED** [review]
LGTM! Thank you 😄

File: src/wp-admin/edit-form-blocks.php [hunk 1]
```diff
@@ -68,8 +68,10 @@ static function( $classes ) {
 	array( rest_get_route_for_post_type_items( 'attachment' ), 'OPTIONS' ),
 	array( rest_get_route_for_post_type_items( 'page' ), 'OPTIONS' ),
 	array( rest_get_route_for_post_type_items( 'wp_block' ), 'OPTIONS' ),
+	array( rest_get_route_for_post_type_items( 'wp_template' ), 'OPTIONS' ),
 	sprintf( '%s/autosaves?context=edit', $rest_path ),
 	'/wp/v2/settings',
```
Comment (SergeyBiryukov) [inline src/wp-admin/edit-form-blocks.php:73]: @Mamaduka Thanks for the PR! Just to clarify, is the `/wp/v2/settings` path without the `OPTIONS` context (line 73) still needed? I don't see any other path loaded both with and without context, but I'm not familiar enough with this part of code 🙂

Comment (TimothyBJacobs) [reply src/wp-admin/edit-form-blocks.php:73]: Yeah, they have different response shapes and are used for different purposes. Would no longer be needed after wordpress/gutenberg#43703, but is for now.

**Comment by SergeyBiryukov** [comment]
Thanks for the PR! Merged in [r54084](https://core.trac.wordpress.org/changeset/54084).

**Pull request closed without merging**